### PR TITLE
Check PAL_VirtualUnwind failure during GC

### PR DESCRIPTION
**Pull request by parjong**
As discussed in #6525, ``PAL_VirtualUnwind`` sometimes fails during GC.

This failure leads to very unpredictable behavior (which is very hard to
analyze).

This commit tries to stop the execution at the point where error
happens.

### CODE REVIEW

**Comment by parjong** [comment]
@jkotas Please take a look

CC @hqueue @chunseoklee

**Comment by jkotas** [comment]
@janvorli Do we want to do the same thing for all other calls to `PAL_VirtualUnwind`?

LGTM otherwise.

**Comment by janvorli** [comment]
@jkotas I had the same thought - I think we do.

File: src/vm/arm/stubs.cpp [hunk 1]
```diff
@@ -544,7 +544,12 @@ void LazyMachState::unwindLazyState(LazyMachState* baseState,
             DacError(hr);
         }
 #else // DACCESS_COMPILE
-        PAL_VirtualUnwind(&ctx, &nonVolRegPtrs);
+        BOOL success = PAL_VirtualUnwind(&ctx, &nonVolRegPtrs);
+        if ( !success )
```
Comment (jkotas) [inline src/vm/arm/stubs.cpp]: Nit - the typical style in CoreCLR is `if (!success)` - not spaces inside parenthesis.

**Comment by jkotas** [comment]
@parjong Could you please make the same change for all other calls to `PAL_VirtualUnwind` - on all platforms?

File: src/vm/arm/stubs.cpp [hunk 2]
```diff
@@ -544,7 +544,12 @@ void LazyMachState::unwindLazyState(LazyMachState* baseState,
...
+        {
+          _ASSERTE(!"unwindLazyState: Unwinding failed");
```
Comment (adityamandaleeka) [inline src/vm/arm/stubs.cpp]: Please use 4 spaces to indent here for consistency.

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**